atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -187,6 +187,7 @@ public Builder socketQueryTimeout(int val){
             config.setTimeBetweenEvictionRunsMillis(TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES));
             config.setNumTestsPerEvictionRun(-1); // Test all idle objects for eviction
             config.setJmxNamePrefix(host);
+            config.setMaxWaitMillis(socketTimeoutMillis);
```

Comment:
borrowObject() just calls borrowObject(config.MaxWaitMillis) internally, so this is the same.

as per tanya's comment, yes, I will implement the retry for this exception.